x-pack/plugins/observability_ai_assistant/server/service/index.ts

```diff
@@ -0,0 +1,152 @@
+    },
+  };
+
+  constructor({ logger, core }: { logger: Logger; core: CoreSetup }) {
+    this.core = core;
+    this.logger = logger;
+  }
+
+  init = once(async () => {
+    try {
+      const [coreStart] = await this.core.getStartServices();
+
+      const esClient = coreStart.elasticsearch.client.asInternalUser;
+
+      await esClient.cluster.putComponentTemplate({
+        create: false,
+        name: this.resourceNames.componentTemplate.conversations,
+        template: conversationComponentTemplate,
+      });
+
+      await esClient.ilm.putLifecycle({
+        name: this.resourceNames.ilmPolicy.conversations,
+        policy: {
+          phases: {
+            hot: {
+              min_age: '0s',
+              actions: {
+                rollover: {
+                  max_age: '90d',
+                  max_primary_shard_size: '50gb',
+                },
+              },
+            },
+          },
+        },
+      });
+
+      await esClient.indices.putIndexTemplate({
+        name: this.resourceNames.indexTemplate.conversations,
+        composed_of: [this.resourceNames.componentTemplate.conversations],
+        create: false,
+        index_patterns: [this.resourceNames.indexPatterns.conversations],
+        template: {
+          settings: {
+            number_of_shards: 1,
+            auto_expand_replicas: '0-1',
+            refresh_interval: '1s',
+          },
+        },
+      });
+
+      const aliasName = this.resourceNames.aliases.conversations;
+
+      await createConcreteWriteIndex({
+        esClient,
+        logger: this.logger,
+        totalFieldsLimit: 10000,
+        indexPatterns: {
+          alias: aliasName,
+          pattern: `${aliasName}*`,
+          basePattern: `${aliasName}*`,
+          name: `${aliasName}-000001`,
+          template: this.resourceNames.indexTemplate.conversations,
+        },
+      });
+    } catch (error) {
+      this.logger.error(`Failed to initialize service: ${error.message}`);
+      this.logger.debug(error);
+      throw error;
+    }
+  });
+
+  async getClient({
+    request,
+  }: {
+    request: KibanaRequest;
+  }): Promise<IObservabilityAIAssistantClient> {
+    const [_, [coreStart, plugins]] = await Promise.all([
+      this.init(),
+      this.core.getStartServices() as Promise<
+        [CoreStart, { security: SecurityPluginStart; actions: ActionsPluginStart }, unknown]
+      >,
+    ]);
+
+    const user = plugins.security.authc.getCurrentUser(request);
+
+    if (!user) {
+      throw Boom.forbidden(`User not found for current request`);
+    }
+
+    const basePath = coreStart.http.basePath.get(request);
+
+    const { spaceId } = getSpaceIdFromPath(basePath, coreStart.http.basePath.serverBasePath);
+
+    return new ObservabilityAIAssistantClient({
+      actionsClient: await plugins.actions.getActionsClientWithRequest(request),
+      namespace: spaceId,
+      esClient: coreStart.elasticsearch.client.asInternalUser,
```

Comment:
Is there any concern with setting this `esClient` with `asInternalUser` to be used by all callers of `ObservabilityAIAssistantClient`?

Reply:
There is not - iirc we discussed this previously, but access to these endpoints is restricted by an access tag that is available to all users that have access to at least one Observability feature. (For 8.10 we will replace this in a follow-up PR with dedicated feature controls.

Reply:
Perfect - thanks @dgieselaar !